specification/blueprint/resource-manager/Microsoft.Blueprint/preview/2018-11-01-preview/assignmentOperation.json

```diff
@@ -0,0 +1,322 @@
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "AssignmentOperation": {
+      "type": "object",
+      "description": "Represents underlying deployment detail for each update to the assignment.",
+      "allOf": [
+        { "$ref": "#/definitions/AzureResourceBase" }
+      ],
+      "properties": {
+        "properties": {
+          "description": "Properties for AssignmentOperation",
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/AssignmentOperationProperties"
+        }
+      }
+    },
+    "AssignmentOperationList":{
+      "type": "object",
+      "description": "List of AssignmentOperation.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "List of AssignmentOperation.",
+          "items": {
+            "$ref": "#/definitions/AssignmentOperation"
+          }
+        },
+        "nextLink": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Link to the next page of results."
+        }
+      }
+
+    },
+    "AssignmentOperationProperties": {
+      "type": "object",
+      "description": "properties of AssignmentOperation.",
+      "properties": {
+        "blueprintVersion": {
+          "type": "string",
+          "description": "The blueprint version used for the assignment operation."
+        },
+        "assignmentState": {
+          "type": "string",
+          "description": "State of this assignment operation."
+        },
+        "timeCreated": {
+          "type": "string",
+          "description": "Create time of this Assignment Operation."
+        },
+        "timeStarted": {
+          "type": "string",
+          "description": "Start time of the underlying deployment."
+        },
+        "timeFinished": {
+          "type": "string",
+          "description": "Finish time of the overall underlying deployments."
+        },
+        "deployments": {
+          "type": "array",
+          "description": "list of jobs in this assignment operation.",
+          "items": {
+            "$ref": "#/definitions/AssignmentDeploymentJob"
+          }
+        }
+      }
+    },
+    "AssignmentDeploymentJob": {
+      "type": "object",
+      "description": "Represents individual job in given assignment operation.",
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "Kind of the job."
+        },
+        "action": {
+          "type": "string",
+          "description": "Name of the action performed in this job."
+        },
+        "jobId": {
+          "type": "string",
+          "description": "Id of this job."
+        },
+        "jobState": {
+          "type": "string",
+          "description": "State of this job."
+        },
+        "result": {
+          "description": "deployment job result.",
+          "$ref": "#/definitions/AssignmentDeploymentJobResult"
+        },
+        "history": {
```

Comment:
This also seems like a good candidate to be modeled as a child action or resource too.
The proposed model assumes the user wants all deployments and the full history of each deployment when retrieving an operation. This gets a lot more cumbersome when they call the list all operations API.

Resource group deployments have a similar scenario and have split these concepts into different nested resources. The deployment contains high level aggregated info and a child resource 'operations' exists for details.

Reply:
in most of the case, it should just have Result, history only appears when you have retry.

each individual retry just have error and resourceCreated, also the resourceCreated would just have {id,Name,Type}, rest of the property will be trimed off.